ARCs/arc-0018.md

```diff
@@ -0,0 +1,601 @@
+
+A Royalty Basis **MUST** be immutable, if an application call is made that would overwrite an existing value it **MUST** fail. 
+
+> The immutability of a royalty basis is important since mutability introduces the possibility for a situation where, after an initial sale, the royalty policy is updated from 1% to 100% for example. This would make any further sales have the full payment amount sent to the royalty recipient and the seller would receive nothing.
+
+The `royalty_receiver` is the address of the account that should receive a partial share of the payment for any [transfer](#transfer) of an asset subject to royalty collection.
+
+The Royalty Collector **MUST** be opted into the full set of assets contained in this list of payment_assets. 
+
+> In the case that an account is not opted into an asset, any transfers where payment is specified for that asset will fail until the account opts into the asset. or the policy is updated.
+
+
+#### Set Payment Asset (REQUIRED):
+
+```
+set_payment_asset(payment_asset: asset, allowed: boolean)
+```
+
+The `payment_asset` argument represents the ASA id that is acceptable for payment. The contract logic **MUST** opt into the asset specified in order to accept them as payment as part of a transfer.
+
+The `allowed` argument is a boolean representing whether or not this asset is allowed.
+
+
+#### Transfer (REQUIRED): 
+
+```
+transfer(
+    royalty_asset: asset, 
+    from: account, 
+    to: account, 
+    royalty_receiver: account, 
+    royalty_asset_amount: uint64, 
+    payment: transaction, 
+    payment_asset: asset, 
+    current_offer_amount: uint64
+)
+```
+
+Transfers the Asset after checking that the royalty policy is adhered to. This call must be sent by the current owner (the `from` parameter)
+
+The `royalty_asset` is the ASA ID to be transferred. It **MUST** have a policy defined prior to attempting a transfer.
+
+The `from` parameter is the account the ASA is transferred from.
+
+The `to` parameter is the account the ASA is transferred to.
+
+The `royalty_receiver` parameter is the account that collects the royalty payment. 
+
+The `royalty_asset_amount` parameter is the number of units of this ASA ID to transfer. The amount **MUST** be less than or equal to the amount [offered](#offer) by the `from` account.
+
+The `payment` parameter is a reference to the transaction that is transferring some asset (ASA or Algos) from the buyer to the Application Address of the Royalty Enforcer. It **MUST** be one of a Payment transaction or Asset Transfer transaction.
+
+The `payment_asset` parameter is used in the case that the payment is being made with some asset rather than with Algos. It **MUST** match the Asset ID of the AssetTransfer payment transaction. 
+
+The `current_offer_amount` parameter is the current amount of the Royalty Asset [offered](#offer) by the `from` account. 
+
+The transfer call **SHOULD** be part of a group with a size of 2 (payment/asset transfer + app call)
```

Comment:
Are we specifying a specific order

Reply:
its specified as part of the abi method signature